File: db-commons/src/main/java/net/blueshell/db/AdvancedController.java
package net.blueshell.db;

import net.blueshell.common.identity.IdentityProvider;

public abstract class AdvancedController<S, AM, SM>  extends IdentityProvider {

    protected final S service;
    protected final AM advancedMapper;
    protected final SM simpleMapper;

    protected AdvancedController(S service, AM advancedMapper, SM simpleMapper) {
        this.service = service;
        this.advancedMapper = advancedMapper;
        this.simpleMapper = simpleMapper;
    }
}

File: api/src/main/kotlin/net/blueshell/api/controller/base/AdvancedController.kt
package net.blueshell.api.controller.base

import net.blueshell.api.auth.IdentityProvider

abstract class AdvancedController<S, AM, SM> protected constructor(
    protected val service: S,
    protected val advancedMapper: AM,
    protected val simpleMapper: SM
) : IdentityProvider()
